core/commands/p2p.go

```diff
@@ -196,6 +202,39 @@ Example:
 	},
 }
 
+// checkPort checks whether target multiaddr contains tcp or udp protocol
+// and whether the port is equal to 0
+func checkPort(target ma.Multiaddr) error {
+	addr := target.String()
+	var sport string
+	if strings.Contains(addr, "tcp") {
```

Comment:
Instead of doing this I would use `ValueForProtocol` from Multiaddr
https://github.com/multiformats/go-multiaddr/blob/master/interface.go#L44

Reply:
@Kubuxu  I had  modified it , could you help me review it again?